azure-devops/vcpkg_windows.txt

```diff
@@ -1,5 +0,0 @@
-boost-build:x86-windows

```

Comment:
i would not remove this file. Think about it as being the file used by CI and by the developer as well to run vcpkg to get all dependencies installed on the dev machine.
The instructions for setting up the machine dev box should indeed state:

run `vcpkg @azure-devops/vcpkg_windows.txt` to locally build all required dependencies.

Reply:
1. I don't think it's right to ask a user to build versions of a library they don't need. At the very least we should split the response file up per triplet.
2. Anyone setting up a vm to run this project will probably also be using our ADO YAML where we use your task so there is no need to have it set up in advance.

Reply:
1. yes, the file could/should be splitted for that reason
2. the Azure DevOps tasks are meant to be run only on the build server (it is possible to run it locally, but over complicated). The ideal solution is that you run vcpkg on CI in the identical way you do on your local dev box: one way to achieve this is to share the response file, so you get something reproducible. The ideal solution is that you do not need any task at all, and you just run the tool directly without any logic in the middle (the task), this is as close as we can get with the current C++ tools. I dream of a package-lock.json for vcpkg too!

Reply:
I remain unconvinced of 2. A user need only run `./vcpkg.exe install boost-math:x86-windows boost-math:x64-windows boost-math:arm-windows boost-math:arm64-windows' if they want to be able to build the STL for all supported targets as documented in our readme. The fact that a pretty short one-liner is sufficient for users to get up and running seems to make the need for response files unnecessary. If we ever end up pulling in more external libraries, which I very much hope we don't, putting those dependencies into a response file may make sense.
The CI is not intended to validate a specific scenario of a user building the STL but rather the correctness of the STL after applying a proposed patch.

Reply:
@cbezault right, for this case the response file is not really needed, as it is simple enough. 

I still dream of a de-facto standard way of installing dependencies for any C++ based project, something like "vcpkg install' (which consume a file that specifies all deps, and produce a lock file).

Reply:
I agree, go bother rob about that 😊